x-pack/legacy/plugins/snapshot_restore/public/app/components/restore_snapshot_form/steps/step_logistics.tsx

```diff
@@ -156,89 +171,146 @@ export const RestoreSnapshotStepLogistics: React.FunctionComponent<StepProps> =
+                                  id="xpack.snapshotRestore.restoreForm.stepLogistics.selectAllIndicesLink"
+                                  defaultMessage="Select all"
+                                />
+                              </EuiLink>
+                            ),
+                        }}
+                      />
+                    ) : null
                   }
                   isInvalid={Boolean(errors.indices)}
                   error={errors.indices}
                 >
-                  <EuiSelectable
-                    allowExclusions={false}
-                    options={indicesOptions}
-                    onChange={options => {
-                      const newSelectedIndices: string[] = [];
-                      options.forEach(({ label, checked }) => {
-                        if (checked === 'on') {
-                          newSelectedIndices.push(label);
-                        }
-                      });
-                      setIndicesOptions(options);
-                      updateRestoreSettings({ indices: [...newSelectedIndices] });
-                      setCachedRestoreSettings({
-                        ...cachedRestoreSettings,
-                        indices: [...newSelectedIndices],
-                      });
-                    }}
-                    searchable
-                    height={300}
-                  >
-                    {(list, search) => (
-                      <EuiPanel paddingSize="s" hasShadow={false}>
-                        {search}
-                        {list}
-                      </EuiPanel>
-                    )}
-                  </EuiSelectable>
+                  {selectIndicesMode === 'list' ? (
+                    <EuiSelectable
+                      allowExclusions={false}
+                      options={indicesOptions}
+                      onChange={options => {
+                        const newSelectedIndices: string[] = [];
+                        options.forEach(({ label, checked }) => {
+                          if (checked === 'on') {
+                            newSelectedIndices.push(label);
+                          }
+                        });
+                        setIndicesOptions(options);
+                        updateRestoreSettings({ indices: [...newSelectedIndices] });
+                        setCachedRestoreSettings({
+                          ...cachedRestoreSettings,
+                          indices: [...newSelectedIndices],
+                        });
+                      }}
+                      searchable
+                      height={300}
+                    >
+                      {(list, search) => (
+                        <EuiPanel paddingSize="s" hasShadow={false}>
+                          {search}
+                          {list}
+                        </EuiPanel>
+                      )}
+                    </EuiSelectable>
+                  ) : (
+                    <EuiFieldText
```

Comment:
What do you think about using the `EuiComboBox` here instead?

Reply:
I think this is a good idea, I'll play around with the placeholder text to make sure the user understands the kind of patterns they can enter